Classes/Emogrifier.php

```diff
@@ -1344,33 +1344,65 @@ function (array $matches) {
         $trimmedLowercaseSelector = trim($lowercasePaddedSelector);
         $xPathKey = md5($trimmedLowercaseSelector);
         if (!isset($this->caches[self::CACHE_KEY_XPATH][$xPathKey])) {
+            $notSelector = $this->extractNotSelector($trimmedLowercaseSelector);
             $finalXpath = '//' . $this->translateCssToXpathPass($trimmedLowercaseSelector);
+
+            if ($notSelector !== '') {
+                $finalXpath = $finalXpath . '[not(' . $this->translateCssToXpathPass($notSelector, true) . ')]';
+            }
+
             $this->caches[self::CACHE_KEY_SELECTOR][$xPathKey] = $finalXpath;
         }
+
         return $this->caches[self::CACHE_KEY_SELECTOR][$xPathKey];
     }
 
     /**
+     * @param string $trimmedLowercaseSelector Whole CSS selector. :not selector part will be removed
+     *
+     * @return string :not selector content or empty string
+     */
+    private function extractNotSelector(&$trimmedLowercaseSelector)
```

Comment:
Please don't use a reference here. We don't want the caller to have their variable modified.